modules/weapons/src/lib.rs

```diff
@@ -1,14 +1,143 @@
+    pub building: f32,
+    /// The damages that the weapon can do to an infantry
+    #[serde(default)]
+    pub infantry: f32,
+    /// The damages that the weapon can do to a vehicle
+    #[serde(default)]
+    pub vehicle: f32,
+    /// The damages that the weapon can do to a tank
+    #[serde(default)]
+    pub armored_vehicle: f32,
+    /// The damages that the weapon can do to a tank
+    #[serde(default)]
+    pub tank: f32,
+    /// The damages that the weapon can do to a helicopter
+    #[serde(default)]
+    pub helicopter: f32,
+    /// The damages that the weapon can do to a plane
+    #[serde(default)]
+    pub plane: f32,
+    /// The damages that the weapon can do to a ship
+    #[serde(default)]
+    pub ship: f32,
+    /// The damages that the weapon can do to a submarine
+    #[serde(default)]
+    pub submarine: f32,
+    /// The damages that the weapon can do to a missile
+    #[serde(default)]
+    pub missile: f32,
+    /// The damages that the weapon can do to a satellite
+    #[serde(default)]
+    pub satellite: f32,
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+/// Define the information that a weapon can have
+///
+/// This structure is used to define the characteristics of a weapon
+///
+/// # Example
+///
+/// ```rs
+/// let weapon = WeaponInformations {
+///   name: "M4A1".to_string(),
+///   caliber: 5.56,
+///   damages: Damages::default(),
+///   speed: 900.0,
+///   range: 500.0,
+///   fire_rate: 800.0
+/// };
+/// ```
+#[derive(Clone, Default, Debug, Serialize, Deserialize, PartialEq, PartialOrd)]
+pub struct WeaponInformations {
+    pub name: String,
+    /// The caliber of the weapon in millimeters
+    #[serde(default)]
+    pub caliber: f32,
+    /// The speed of the projectile in meters per second
+    #[serde(default)]
+    pub speed: Speed,
+    /// The range in kilometers
+    #[serde(default)]
+    pub range: f32,
+    /// The country reference of the weapon, it's used to know which country can use the weapon
+    ///
+    /// TODO Use a custom type instead of a String
+    pub country_reference: String
+}
+#[cfg(feature = "load_configuration")]
+pub mod loader {
+    use std::{fs, io};
+    use std::collections::HashMap;
+    use std::path::PathBuf;
+    use toml::Value;
+    use crate::{Damages, WeaponInformations};
+    use crate::missiles::Missile;
+    use crate::shells::Shell;
 
-    #[test]
-    fn it_works() {
-        let result = add(2, 2);
-        assert_eq!(result, 4);
+    pub(crate) fn parse_weapons_information(t: &Value) -> Result<WeaponInformations, toml::de::Error> {
+        t.clone().try_into()
     }
-}
+
+    pub(crate) fn parse_damages(t: &Value) -> Result<Damages, toml::de::Error> {
+        t.clone().try_into()
+    }
+
+    pub type WeaponID = String;
+
+    /// Store all the weapons
+    ///
+    /// This structure is used to store all the weapons that are loaded from the files in the weapons folder
+    #[derive(Default, Clone, Debug)]
+    pub struct WeaponsStore {
```

Comment:
we could also add torpedoes, infantry weapons...

Reply:
Yeah why not, this code is versatile, it means that we can add a lot of other weapons